### PR TITLE
Terror event spawns now pick players randomly

### PR DESCRIPTION
**Pull request by Kyep**
## What Does This PR Do
Previously, the terror spiders major event spawned its spiders, then offered each to ghosts individually, with the first player to click a spider or its prompt getting control of that spider.
Now, the terror spiders event polls eligible ghosts for 30 seconds, then assigns spiders randomly to eligible opted-in players. Only after its picked who will be what spider, does it actually spawn any spiders.
This means players who accept quickly no longer have advantage on becoming initially spawned terror spiders.
It also means that if not enough people sign up to play terrors during an event, admins are warned and fewer than normal spiders are spawned.

## Why It's Good For The Game
Normally, if terrors are being spawned there's generally enough of them for them to be handled on a first-come, first-serve basis.
However, it is possible for the terror spider spawn event to choose to spawn, say, only a single prince of terror.
When that happens, its probably best for selection there to be entirely random, as its likely to be the only terror in the round at all.

## Note
Due to multiplayer issues, this will require a TM to fully test.
This only applies to the initial 1-5 spiders directly spawned by the event. It does NOT apply to any terror who grows up from a spiderling. Don't ask me to apply it there too. I may or may not do that in future, but I want to see how this goes before I even consider that.

## Changelog
:cl: Kyep
tweak: terror spiders spawned directly by event now get offered to all eligible ghosts for 30 seconds, and pick players randomly from that pool.
/:cl:

### CODE REVIEW

File: code/modules/events/spider_terror.dm
Comment (SteelSlayer) [inline code/modules/events/spider_terror.dm:46]: Why is this necessary? What else needs to run before the code below?

Comment (Kyep) [reply code/modules/events/spider_terror.dm:46]: Without it, the whole thing breaks. The event loops and it generates incorrect popups.

Comment (AffectedArc07) [inline code/modules/events/spider_terror.dm:46]: I am aware that killing the `spawn()` here will break things. **However**, I am going to ask you offload this to another proc and use `INVOKE_ASYNC` to do the work instead of wrapping all of this inside a `spawn()`

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]

**Comment by Spacemanspark** [comment]
Just to be clear, does this only impact the initial wave of terrors? Or does this impact all terror spawns?

**Comment by cluwneythescourge** [comment]
thank goodness for this!

**Review by AffectedArc07 — APPROVED** [review]

**Comment by Kyep** [comment]
@Spacemanspark only the initial spiders spawned by the event. I won't be applying it to all spiders in this PR (and will likely object if anyone tries). That's something for me to consider in another PR, later, once I'm sure I'm happy with the effect this has.

**Review by Dumbdumn5 — APPROVED** [review]
Though I would personally support this for all spiders, this is a step in the right direction, RNG is a glorious thing and so is giving everyone a chance to actually play a midround without dedicating themselves to hunting the things down and slapping the thing as soon as it shows up. :+1:

**Comment by Kyep** [comment]
Please don't merge this yet, I want to do a TM of it on the live server before its merged.
I realize this is probably excessively cautious, I've tested it several times, but I saw some really bad timing issues when spawn() was not used, and while I have tested it several times with INVOKE_ASYNC I want a live server test before full merge just to be doubly sure.

**Comment by AffectedArc07** [comment]
TM'd with 0 issues

**Pull request merged**
